terminal/plugins/org.eclipse.tm.terminal.control/src/org/eclipse/tm/internal/terminal/control/ITerminalListener.java

```diff
@@ -30,6 +30,7 @@ public interface ITerminalListener {
 	/**
 	 * Set the title of the terminal.
 	 * @param title
+	 * @param requestor Who requests update.
 	 */
-	void setTerminalTitle(String title);
```

Comment:
This is somewhat frustrating (and not something I realized until now). 

When this code was converted to API in dea02eb9f98dbdbcea414adf68894e8af04f23f1 it seems that there was a decision not to rename these packages. So even though they have "internal" in their package names, they are API of the bundle:

https://github.com/eclipse-cdt/cdt/blob/b2666eb57819b3e22b22bf05f586caa95dc34df4/terminal/plugins/org.eclipse.tm.terminal.control/META-INF/MANIFEST.MF#L15-L18

That means we need to [evolve this API](https://github.com/eclipse-cdt/cdt/blob/main/POLICY.md#api) in a non-breaking way.

Do you think you can evolve the API? Do you know about the API rules?

Reply:
Did not know about that. For now I do not have better idea how to implement this fix. Need to think more about it. Do you have any ideas?

Reply:
Well, we can try to pass complete ANSI string in https://github.com/eclipse-cdt/cdt/blob/af7643e0d0b7fd723ea1aaf00bde78ea9eeb640d/terminal/plugins/org.eclipse.tm.terminal.control/src/org/eclipse/tm/internal/terminal/emulator/VT100Emulator.java#L489 and analyze it in updateTitle for leading "0;" symbols and then consider it as passed using ANSI command but I do not think it will be a robust solution. Probably setting complete "\e]0;" prefix and passing whole string to setTerminalTitle and further string analysis in updateTitle will be more robust. This will allow us to keep existing implementation of setTerminalTitle method.

Reply:
I'll try to throw a prototype together of how to write your code using an evolved API.

Reply:
I force pushed with my proposal on how to handle it - there is still some reviewing and updating you need to do, but see https://github.com/eclipse-cdt/cdt/pull/508/commits/18634bb4971fcf78eee467ebf5e173ef023a011b for a summary of what I propose.